charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml

```diff
@@ -65,6 +65,9 @@ spec:
             {{- end }}
             - --kubelet-endpoints={{ .Values.prometheusOperator.kubeletEndpointsEnabled }}
             - --kubelet-endpointslice={{ .Values.prometheusOperator.kubeletEndpointSliceEnabled }}
+            {{- if .Values.prometheusOperator.kubeletService.labels }}
+            - --labels={{ .Values.prometheusOperator.kubeletService.labels }}
```

Comment:
As discussed with @QuentinBisson earlier `prometheusOperator.kubeletService.labels` assume, that only the `kubeletService` get additional labels. However all resources created by the operator. Reference `kubeletService` is not correct here.

I would recommend a more generic approach, e.g.:

https://github.com/prometheus-community/helm-charts/blob/f828f547380ce33ea5bd7d7b483bb573a8df1a70/charts/kube-state-metrics/templates/deployment.yaml#L71-L73

where users can pass any CLI arguments as they want.

Reply:
that make sense; Removed that change

Reply:
@kranthikirang just to prevent an missunderstanding, but the extraArgs needs to be implemented for the operator

Reply:
`{{- if .Values.prometheusOperator.extraArgs }}
{{- range .Values.prometheusOperator.extraArgs }}
- {{ . }}
{{- end }}
{{- end }}`

Modified and tested. Please check now

Reply:
Also added an example in values file